packages/cadl-ranch-specs/http/azure/client-generator-core/access/main.tsp

```diff
@@ -0,0 +1,191 @@
+  @get
+  @global.Azure.ClientGenerator.Core.access(global.Azure.ClientGenerator.Core.Access.public)
+  op publicDecoratorInPublic(@query name: string): PublicDecoratorModelInPublic;
+}
+
+@route("/internal")
+@global.Azure.ClientGenerator.Core.operationGroup
+@scenario
+@scenarioDoc("""
+This scenario contains internal operations. All should be generated but not exposed.
+Expected query parameter: name=<any string>
+Expected response body:
+```json
+{
+  "name": <any string>
+}
+```
+""")
+namespace Internal {
+  @doc("Used in an internal operation, should be generated but not exported.")
+  model NoDecoratorModelInInternal {
+    name: string;
+  }
+
+  @doc("Used in an internal operation, should be generated but not exported.")
+  @global.Azure.ClientGenerator.Core.access(global.Azure.ClientGenerator.Core.Access.internal)
+  model InternalDecoratorModelInInternal {
+    name: string;
+  }
+
+  @doc("Used in an internal operation but with public decorator, should be generated and exported.")
+  @global.Azure.ClientGenerator.Core.access(global.Azure.ClientGenerator.Core.Access.public)
+  model PublicDecoratorModelInInternal {
+    name: string;
+  }
+
+  @route("/noDecoratorInInternal")
+  @get
+  @global.Azure.ClientGenerator.Core.access(global.Azure.ClientGenerator.Core.Access.internal)
+  op noDecoratorInInternal(@query name: string): NoDecoratorModelInInternal;
+
+  @route("/internalDecoratorInInternal")
+  @get
+  @global.Azure.ClientGenerator.Core.access(global.Azure.ClientGenerator.Core.Access.internal)
+  op internalDecoratorInInternal(@query name: string): InternalDecoratorModelInInternal;
+
+  @route("/publicDecoratorInInternal")
+  @get
+  @global.Azure.ClientGenerator.Core.access(global.Azure.ClientGenerator.Core.Access.internal)
+  op publicDecoratorInInternal(@query name: string): PublicDecoratorModelInInternal;
+}
+
+@route("/shared")
+@global.Azure.ClientGenerator.Core.operationGroup
+@scenario
+@scenarioDoc("""
+This scenario contains two operations, one public, another internal. The public one should be generated and exported while the internal one should be generated but not exposed.
+Expected query parameter: name=<any string>
+Expected response body:
+```json
+{
+  "name": <any string>
+}
+```
+""")
+namespace Shared {
+  @doc("Used by both public and internal operation. It should be generated and exported.")
+  model SharedModel {
+    name: string;
+  }
+
+  @route("/public")
+  @get
+  @global.Azure.ClientGenerator.Core.access(global.Azure.ClientGenerator.Core.Access.public)
+  op public(@query name: string): SharedModel;
+
+  @route("/internal")
+  @get
+  @global.Azure.ClientGenerator.Core.access(global.Azure.ClientGenerator.Core.Access.internal)
+  op internal(@query name: string): SharedModel;
+}
+
+@route("/relative")
+@global.Azure.ClientGenerator.Core.operationGroup
+@scenario
+@scenarioDoc("""
+This scenario contains internal operations. All should be generated but not exposed.
+""")
+namespace Relative {
```

Comment:
What is Relative? It is just all internal models. NestedModel?

Reply:
It just means the models here are related, like A is B's property, A is B's base model, etc. I didn't figure out a better name.